recipes/perl/assets/create_assetGroup.pl

```diff
@@ -0,0 +1,209 @@
+    }
+    else {
+        print "Request failed with status code: ", $resp->code, "\n";
+    }
+    return $token;
+}
+
+# subroutine to get assets
+sub create_assetGroup {
+    my $url = "$base_url/asset-service/queries";
+
+    my $asset_data = qq({
+  "data": {
+    "type": "query",
+    "attributes": {
+      "queryName": "create-or-update-assets",
+      "workloads": ["vmware"],
+      "parameters": {
+        "objectList": [
+        {
+            "correlationId": "cor1223",
+            "type": "vmwareGroupAsset",
+            "assetGroup": {
+              "description": "sampleDescription",
+              "assetType": "vmGroup",
+              "filterConstraint": "sampleFilterConstaint",
+              "oDataQueryFilter": "commonAssetAttributes/displayName eq 'test125478'",
+              "commonAssetAttributes": {
+                "displayName": "sampleGroup249vbgt",
+                "workloadType": "vmware",
+                "protectionCapabilities": {
+                  "isProtectable": "YES",
+                  "isProtectableReason": "sampleReason",
+                  "isRecoverable": "NO",
+                  "isRecoverableReason": "sampleReason"
+                },
+                "detection": {
+                  "detectionMethod": "MANUAL"
+                }
+              }
+            }
+          }
+        ]
+      }
+    }
+  }
+});
+
+    my $req = HTTP::Request->new(POST => $url);
+    $req->header('Content-Type' => $content_type);
+    $req->header('Authorization' => $token);
+    $req->content($asset_data);
+
+    print "\n\n**************************************************************";
+    print "\n\n Making POST Request to create assetGroup\n\n";
+
+    my $resp = $ua->request($req);
+ 
+        my $message = $resp->decoded_content;
+        print "Create vmware Asset Group succeeded with status code: ", $resp->code, "\n\n\n";
+        print "=======================\n";
+
+        my $json = decode_json($message);
+        my $id = $json->{'data'}->{'id'};
+        print "id: ", $id, "\n\n\n";
+        
+        my $url = "$base_url/asset-service/queries/$id";
+        my $req = HTTP::Request->new(GET => $url);
+        $req->header('Accept' => $content_type);
+        $req->header('Authorization' => $token);
+
+        print "\n\n**************************************************************";
+        print "\n\n Making GET Request for the QueryID response\n\n";
+
+        my $resp = $ua->request($req);
+        my $message = $resp->decoded_content;
+        
+        print "Create vmware Asset Group query id : ", $resp->code, "\n\n\n";
+
+        my $json = decode_json($message);
+        my @response = @{$json->{'data'}};
+        my $response = $_;
+        my @workItem = @{$response->{'attributes'}->{'workItemResponses'}};
+        my $workitem = $_;
+        my $status = $workItem->{'statusDetails'}->{'message'};
+
+        if ($status = created) {
```

Comment:
This assigns the bareword `created` to `$status`, them tested whether it's true (which it always is). You probably meant to use the `==` operator and quotes, although recall that the comparison operator for strings is `eq`.

```suggestion
        if ($status eq 'created') {
```

You should `use strict` to help avoid this mistake.

Reply:
Done.